### DISCUSSION TITLE
Multi-Tenancy?

### DISCUSSION THREAD

**Discussion by mansellan** [SigNoz/signoz#1471] [Ideas]
Hi,

Firstly many thanks for the awesome work! We're in the market for an APM, and SigNoz ticks a lot of boxes for us.

One question I have though is: are there any plans to support multi-tenancy in the Enterprise edition? We currently run a single Kubernetes cluster for all our non-production environments, so it would be ideal if we could access telemetry for each one through a different URL.

**Comment by pranay01** [comment]
hey @mansellan welcome to the SigNoz community :) 

Are you looking for multi tenancy in self hosted Enterprise version or in the SigNoz cloud version?

If your use case is to see different environments easily, you can even today pass environments as attributes and filter based on that.

![Screenshot 2022-08-03 at 10 27 28 PM](https://user-images.githubusercontent.com/504541/182666130-42d71182-45ed-497a-a318-5ba3aa17917d.png)

If this doesn't serve your use case, can you share more on why would you need telemetry for each environment in different URLs? Are these different projects or different teams in your org?

**Reply by mansellan** [reply]
Aha perfect, attribute filtering should do the trick nicely, many thanks!

We're self-hosting, and will be looking into Enterprise soon (need the LDAP integration)

**Reply by mansellan** [reply]
So attribute filtering does work as expected, but it's a little inconvenient - you have to manually filter each view each time you switch between areas of the UI. It would be awesome if we could set "global" attribute filters somehow.

**Comment by mdella-nutanix** [comment]
In our case, we have multiple projects that typically do not have access to one another. We want to run Signoz as a utility for the different teams yet will only have one team managing the infrastructure for the app. Without multi-tenancy (ie, LDAP groups or manual isolation) this will basically fall off of our selection list for Nutanix. Since we're in a review mode, if this doesn't have any roadmap, then we'll have to drop the product.

**Reply by ankitnayan** [reply]
This is an enterprise feature. Did you already have a discussion about that? I think the team should be able to roll this out for enterprise customers

**Reply by mdella-nutanix** [reply]
Have not had discussions around this also don't have a contact over the enterprise version (not ability to test at the moment, we're testing the community version)
